app/routes/campaign.js

```diff
@@ -17,8 +17,11 @@ module.exports = function (router) {
     router.post('/campaign/create', (request, response) => {
         if(request.user) {
             campaignPromise = campaignController.saveCampaign(request);
-            campaignPromise.then(function(campaign) {
-                response.send(campaign._id);
+            campaignPromise.then(function() {
+                return campaignController.findCampaignByTitleAndUser(request.body.title, request.user.id);
+            })
+            .then(function(result) {
+                response.send(result[0]._id);
```

Comment:
Ok, I'm not able to replicate this issue, but since neither of us have seen the issue with the strategy in this PR, lets go with this one.  This might be a good opportunity to start talking about testing. A good way to try to make this issue bubble up might be to fire off 100-1k saveCampaign calls and see which ones don't return a newly created campaign with shortId.